src/common/browserUtils.ts

```diff
@@ -0,0 +1,14 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+let inBrowser = false
+/** Set the value of if we are in the browser. Impacts {@link isInBrowser}. */
+export function setInBrowser(value: boolean) {
+    inBrowser = value
+}
+/** Return true if we are running in the browser, false otherwise. */
+export function isInBrowser() {
```

Comment:
These "in a browser?" functions could live in `src/shared/extensionUtilities.ts`.  Compare `getIdeProperties()` which lives in `extensionUtil.ts`.

Reply:
Also, [we don't need /common/](https://github.com/aws/aws-toolkit-vscode/pull/3785#discussion_r1309314009) if we have /shared/, right? Instead, we can incrementally _move_ browser-only or node-only things _out of_ /shared/, so eventually /shared/ will have only "platform-agnostic" code in it.

Reply:
The issue I found was that with any import of a /shared/ module one of the transitive dependencies caused issues and it was a hassle to chase it down. 

- it will become a whack-a-mole if we import something from shared but some separate code on the module causes issues and we need to find all node/browser specific code and move it out
- On the other hand, if we use some code from `src/shared/` and realize it is platform agnostic, we can move it to `src/common`